cpp/source/deal.II/electrochemical_physics.templates.h

```diff
@@ -0,0 +1,257 @@
+      for (unsigned int i = 0; i < dofs_per_cell; ++i)
+      {
+        for (unsigned int j = 0; j < dofs_per_cell; ++j)
+        {
+          // Mass matrix terms
+          double const mass_matrix_val =
+              specific_capacitance_values[q] *
+              (fe_values[solid_potential].value(i, q) *
+                   fe_values[solid_potential].value(j, q) -
+               fe_values[solid_potential].value(i, q) *
+                   fe_values[liquid_potential].value(j, q) -
+               fe_values[liquid_potential].value(i, q) *
+                   fe_values[solid_potential].value(j, q) +
+               fe_values[liquid_potential].value(i, q) *
+                   fe_values[liquid_potential].value(j, q)) *
+              fe_values.JxW(q);
+          cell_mass_matrix(i, j) += mass_matrix_val;
+          cell_system_matrix(i, j) +=
+              mass_matrix_val +
+              time_step *
+                  // Stiffness matrix terms
+                  (solid_phase_diffusion_coefficient_values[q] *
+                       (fe_values[solid_potential].gradient(i, q) *
+                        fe_values[solid_potential].gradient(j, q)) +
+                   liquid_phase_diffusion_coefficient_values[q] *
+                       (fe_values[liquid_potential].gradient(i, q) *
+                        fe_values[liquid_potential].gradient(j, q)) +
+                   faradaic_reaction_coefficient_values[q] *
+                       ((fe_values[solid_potential].value(i, q) *
+                         fe_values[solid_potential].value(j, q)) -
+                        (fe_values[liquid_potential].value(i, q) *
+                         fe_values[solid_potential].value(j, q)) -
+                        (fe_values[solid_potential].value(i, q) *
+                         fe_values[liquid_potential].value(j, q)) +
+                        (fe_values[liquid_potential].value(i, q) *
+                         fe_values[liquid_potential].value(j, q)))) *
+                  fe_values.JxW(q);
+        }
+      }
+
+    // Fill in the global matrices.
+    cell->get_dof_indices(local_dof_indices);
+    this->constraint_matrix.distribute_local_to_global(
+        cell_system_matrix, cell_rhs, local_dof_indices, this->system_matrix,
+        this->system_rhs, inhomogeneous_bc);
+    // TODO is this correct? Do we have constraints on the mass matrix?
+    for (unsigned int i = 0; i < dofs_per_cell; ++i)
+      for (unsigned int j = 0; j < dofs_per_cell; ++j)
+        this->mass_matrix.add(local_dof_indices[i], local_dof_indices[j],
+                              cell_mass_matrix(i, j));
+  }
+
+  // Add the null space. This wouldn't be necessary if we were using a
+  // hp::DoFHandler object instead of a DoFHandler object but
+  // hp::DoFHandler does not work with MPI.
+  dealii::types::global_dof_index const n_dofs = this->dof_handler->n_dofs();
+  for (unsigned int i = 0; i < n_dofs; ++i)
+    if (std::fabs(this->system_matrix.diag_element(i)) < 1e-100)
```

Comment:
I thought that in C++11 <cmath> had overloads for all types with ``std::abs``.
Not a big deal, just wondering why you went for ``fabs`` instead.